include/swift/AST/Attr.h

```diff
@@ -1709,6 +1709,7 @@ class ImplementsAttr : public DeclAttribute {
                                 DeclName MemberName);
 
   ProtocolDecl *getProtocol(DeclContext *dc) const;
+  std::optional<ProtocolDecl *> getCachedProtocol(DeclContext *dc) const;
```

Comment:
Worth adding a similar doc comment to the others?

Reply:
Done.